languages/swift/exercises/concept/get-organized/.docs/instructions.md

```diff
@@ -0,0 +1,15 @@
+You are an internationally renowned organizing guru; your books and lectures have helped people around the world to organize and declutter their lives. To do this you have developed a number of tools to help organize all sorts of things. You have a tool to organize `Int`s a tool to organize `String`s a tool to organize `Double`s and on and on. There's only one problem… you have so many tools, you can't keep them organized.
+
+What you need to do in order to declutter your life is write a single tool that can organize _any_ type of thing. You have two tasks to help you do just that.
+
+##1. Swap two elements in any array
+
+For this task, you will need to write a function, `swapAt(_:_:_:)` where the first parameter is an inout array of elements of any type and the second and third parameters are `Int` indices of two elements in the array. If the two indices are valid indices, the function will swap the values at the two indices, returning the modified array via the inout parameter. If either index is invalid, the array should remain unchanged.
+
+##2. Bubble sort any array
```

Comment:
```suggestion
## 2. Bubble sort any array
```

Reply:
I'm not entirely sure about the bubble sorting. I _think_ we discussed somewhere that we didn't want concept exercises to implement algorithms (practice exercises are perfect for that). Is that correct @iHiD?

Reply:
I'll wait with the rest of the review when @iHiD replied to the above comment.

Reply:
If we want to use a Bubble Sort, we need to be a little more explicit about how to solve it (ie, give the algorithm away). The reason for this is that we want Concept Exercises to be solvable by someone familiar in a language within 5mins, and having to work out algorithms can extend that. Algorithms can also feel like a real turnoff to lots of people (myself included), and intimidating to people who feel insecure as they don't have CS degrees, so it's important to emphasise that they're not being tested on the algorithm itself, just the using Swift (the fun bit!).

> For this task, you are you going to implement a [Bubble Sort][bubble-sort] in Swift. This a simple algorithm that repeatedly steps through a list, comparing adjacent elements and swapping them if they are in the wrong order, until the list is sorted.
> 
> You will need to write a function `bubbleSort(_:swapIf:)`. The first parameter of `bubbleSort` is an inout array of elements of any type. The second parameter is a function that takes two values of the same type as the elements of the array and returns `true` if the first value should appear _after_ the second element in the array.  The function should use the `swapIf` parameter to determine if two elements need to be swapped and `swapAt` from you first task to swap the elements when needed.

Sidenote: Is the function expecting the user to write the loop to do this? That wasn't clear to me and we should be explicit about it if so. The easiest way is to give example of the input and output.e.g.

```swift
bubbleSort([1,3,2,4])
=> [1,2,3,4]
```